### ISSUE TITLE
The idea of a neural network using TensorFlow.js for solving classification tasks.

### ISSUE THREAD

**Issue by RuslanKryvtsov** [RuslanKryvtsov/JS#1]
1) <!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Language Detection Example</title>
  <script src="https://cdn.jsdelivr.net/npm/@tensorflow/tfjs@3.9.0/dist/tf.min.js"></script>
</head>
<body>
  <div id="output"></div>
  <script>
    // Імпортуємо необхідні бібліотеки
    const tf = window.tf;
    
    // Зареєструємо ядра для backend 'cpu'
    tf.setBackend('cpu');

    // Визначаємо архітектуру нейромережі
    const model = tf.sequential();
    model.add(tf.layers.dense({ units: 10, inputShape: [6], activation: 'relu' }));
    model.add(tf.layers.dense({ units: 2, activation: 'softmax' }));

    // Компілюємо модель
    model.compile({ loss: 'categoricalCrossentropy', optimizer: 'adam', metrics: ['accuracy'] });

    // Створюємо тренувальні дані
    const trainData = tf.tensor2d([
      [1, 0, 0, 0, 0, 0],  // "apple" (англійська)
      [0, 1, 0, 0, 0, 0],  // "banana" (англійська)
      [0, 0, 1, 0, 0, 0],  // "cat" (англійська)
      [0, 0, 0, 1, 0, 0],  // "яблуко" (українська)
      [0, 0, 0, 0, 1, 0],  // "банан" (українська)
      [0, 0, 0, 0, 0, 1],  // "кіт" (українська)
    ], [6, 6]);

    const trainLabels = tf.tensor2d([
      [1, 0],  // Англійська
      [1, 0],  // Англійська
      [1, 0],  // Англійська
      [0, 1],  // Українська
      [0, 1], // Українська
      [0, 1] , // Українська
      ]);
// Тренуємо модель
async function trainModel() {
  await model.fit(trainData, trainLabels, { epochs: 10 });
  console.log('Модель навчена!');
}

// Тестуємо модель на нових даних
function testModel() {
  const testData = tf.tensor2d([[1, 0, 0, 0, 0, 0], [0, 0, 0, 1, 0, 0]], [2, 6]);
  const predictions = model.predict(testData);
  const predictedLabels = tf.argMax(predictions, axis=1).arraySync();
  const languages = ['Англійська', 'Українська'];

  const outputElement = document.getElementById('output');
  outputElement.innerHTML = '';
  for (let i = 0; i < predictedLabels.length; i++) {
    const predictedLabel = predictedLabels[i];
    const predictedLanguage = languages[predictedLabel];
    const resultElement = document.createElement('p');
    resultElement.innerText = `Слово ${i + 1} відноситься до мови: ${predictedLanguage}`;
    outputElement.appendChild(resultElement);
  }
}

// Запускаємо тренування моделі та викликаємо тестування
async function run() {
  await trainModel();
  testModel();
}

run();

</script> 
</body> 
</html>

2)  // Використовуємо backend 'cpu'
tf.setBackend('cpu');

// Тестуємо модель на нових даних
function testModel() {
  const testData = tf.tensor2d([[0, 0, 0, 1, 0, 0], [0, 1, 0, 0, 0, 0], [0, 0, 1, 0, 0, 0], [0, 0, 0, 0, 1, 0], [1, 0, 0, 0, 0, 1]], [5, 6]);
  const predictions = model.predict(testData);
  const predictedLabels = tf.argMax(predictions, axis=1).arraySync();
  const languages = ['Англійська', 'Українська'];

  const outputElement = document.getElementById('output');
  outputElement.innerHTML = '';
  for (let i = 0; i < predictedLabels.length; i++) {
    const predictedLabel = predictedLabels[i];
    const predictedLanguage = languages[predictedLabel];
    const resultElement = document.createElement('p');
    resultElement.innerText = `Слово ${i + 1} відноситься до мови: ${predictedLanguage}`;
    outputElement.appendChild(resultElement);
  }
}

// Запускаємо тренування моделі та викликаємо тестування
async function run() {
  // Тренуємо модель
  await trainModel();

  // Компілюємо модель
  model.compile({ loss: 'binaryCrossentropy', optimizer: 'adam', metrics: ['accuracy'] });

  // Тестуємо модель
  testModel();
}

run();

Необхідно покращити точність роботи нейромережі. Для цього:

1)	Збільшити обсяг тренувальних даних, включаючи більше різних слів з англійської та української мов;
2)	Переконатися, що тренувальні дані відображають різноманітність слів  в кожній мові;
3)	Перевірити архітектуру та параметри моделі нейромережі, можливо, варто розглянути інші моделі або змінити параметри навчання для досягнення кращої точності.
